minidump-processor/src/op_analysis.rs

```diff
@@ -0,0 +1,733 @@
+            data.expected_size = 2;
+            access_test(&data);
+
+            // mov eax, [rbx + rcx]
+            data.bytes = &[0x8b, 0x04, 0x0b];
+            data.expected_size = 4;
+            access_test(&data);
+
+            // mov rax, [rbx + rcx]
+            data.bytes = &[0x48, 0x8b, 0x04, 0x0b];
+            data.expected_size = 8;
+            access_test(&data);
+        }
+
+        #[test]
+        fn test_base_index_disp() {
+            // mov al, [rcx + r9 + 16]
+            let mut data = AccessTestData {
+                bytes: &[0x42, 0x8a, 0x44, 0x09, 0x10],
+                regs: &[("rcx", 0x4000), ("r9", 0x2000)],
+                expected_size: 1,
+                expected_addresses: &[0x6010],
+            };
+            access_test(&data);
+
+            // mov ax, [rcx + r9 + 16]
+            data.bytes = &[0x66, 0x42, 0x8b, 0x44, 0x09, 0x10];
+            data.expected_size = 2;
+            access_test(&data);
+
+            // mov eax, [rcx + r9 + 16]
+            data.bytes = &[0x42, 0x8b, 0x44, 0x09, 0x10];
+            data.expected_size = 4;
+            access_test(&data);
+
+            // mov rax, [rcx + r9 + 16]
+            data.bytes = &[0x4a, 0x8b, 0x44, 0x09, 0x10];
+            data.expected_size = 8;
+            access_test(&data);
+
+            // mov rax, [rcx + r9 - 16]
+            data.bytes = &[0x4a, 0x8b, 0x44, 0x09, 0xf0];
+            data.expected_size = 8;
+            data.expected_addresses = &[0x5ff0];
+            access_test(&data);
+        }
+
+        #[test]
+        fn test_index_scale_disp() {
+            // mov al, [r13 * 8 + 0x100000]
+            let mut data = AccessTestData {
+                bytes: &[0x42, 0x8a, 0x04, 0xed, 0x00, 0x00, 0x10, 0x00],
+                regs: &[("r13", 0x1000)],
+                expected_size: 1,
+                expected_addresses: &[0x108000],
+            };
+            access_test(&data);
+
+            // mov ax, [r13 * 8 + 0x100000]
+            data.bytes = &[0x66, 0x42, 0x8b, 0x04, 0xed, 0x00, 0x00, 0x10, 0x00];
+            data.expected_size = 2;
+            access_test(&data);
+
+            // mov eax, [r13 * 8 + 0x100000]
+            data.bytes = &[0x42, 0x8b, 0x04, 0xed, 0x00, 0x00, 0x10, 0x00];
+            data.expected_size = 4;
+            access_test(&data);
+
+            // mov rax, [r13 * 8 + 0x100000]
+            data.bytes = &[0x4a, 0x8b, 0x04, 0xed, 0x00, 0x00, 0x10, 0x00];
+            data.expected_size = 8;
+            access_test(&data);
+
+            // mov rax, [r13 * 8 - 0x100000]
+            data.bytes = &[0x4a, 0x8b, 0x04, 0xed, 0x00, 0x00, 0xf0, 0xff];
+            data.expected_size = 8;
+            data.expected_addresses = &[0xfffffffffff08000];
+            access_test(&data);
+        }
+
+        #[test]
+        fn test_base_index_scale() {
+            // mov al, [r12 + r14 * 2]
+            let mut data = AccessTestData {
+                bytes: &[0x43, 0x8a, 0x04, 0x74],
```

Comment:
I think what you're doing here is cleaner but just in case you're unaware, this is also a valid way to declare a byte array/slice: `b"\x43\x8a\x04\x74"`

Reply:
Now that you mention it, I have seen that before but it was buried away somewhere deep. I think I still like this better, TBH